code/game/objects/structures/crates_lockers/closets/cardboardbox.dm

```diff
@@ -62,3 +64,17 @@
 			for(var/mob/M in viewers(src))
 				M.show_message("<span class='notice'>\The [src] has been cut apart by [user] with \the [WC].</span>", 3, "You hear cutting.", 2)
 			qdel(src)
+			return
+		if(istype(W, /obj/item/weapon/storage/fancy/crayons)) /////addendum by flimflamm
+			var/decalselection = input("Please select a decal") in list( "cancel", "atmospherics", "bartender", "barber", "blueshield",	"brig_physician", "captain",
+				"cargo", "chief_engineer",	"chaplain",	"chef", "chemist", "civilian", "clown", "cmo", "coroner", "detective", "engineering", "genetics", "hop",
+				"hos", "hydroponics", "internal_affairs_agent", "janitor",	"magistrate", "mechanic", "medical", "mime", "mining", "nt_representative", "paramedic", "pod_pilot",
+				"prisoner",	"research_director", "security", "syndicate", "therapist", "virology", "warden", "xenobiology")
+			if(decalselection == "cancel")
+				return
```

Comment:
Once you're canceling as null, the `if` for this return can just be `if(!decalselection)`.

> also, is there a standard distance check proc/set of sanity checks i should be using?

The standard adjacency check is `sometThing.Adjacent(otherThing)`. You'll want to make sure the box is adjacent to the user, and the crayons are still in their active hand (`W == user.get_active_hand()`). You'll also want to check `user.incapacitated()` - if true, the user is no longer in any condition to finish the task.